packages/compiler/core/checker.ts

```diff
@@ -4016,57 +4022,67 @@ export function createChecker(program: Program): Checker {
     if (filter) {
       model = filterModelProperties(model, filter);
     }
-    while (true) {
-      if (model.name) {
-        // named model
-        return model;
-      }
 
-      // We would need to change the algorithm if this doesn't hold. We
-      // assume model has no inherited properties below.
-      compilerAssert(!model.baseModel, "Anonymous model with base model.");
+    if (model.name) {
+      // named model
+      return model;
+    }
 
-      if (model.properties.size === 0) {
-        // empty model
-        return model;
-      }
+    // We would need to change the algorithm if this doesn't hold. We
+    // assume model has no inherited properties below.
+    compilerAssert(!model.baseModel, "Anonymous model with base model.");
```

Comment:
Incidentally, I had it backwards when I worried when we discovered you can extend an alias to extend an anonymous type. The assertion here is that the anonymous type has no base type, which is different, and I believe still impossible.